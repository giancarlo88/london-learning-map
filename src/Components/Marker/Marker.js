import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const StyledMarker = styled.div`
  background-image: url(/marker.svg);
  background-size: cover;
  height: 30px;
  width: 20px;
`

const Marker = ({ lat, lng, index, handleClick }) => (
  <StyledMarker lat={lat} lng={lng} index={index} onClick={() => handleClick(index)}/>
)

Marker.propTypes = { 
  index: PropTypes.number.isRequired, 
  handleClick: PropTypes.func.isRequired,
  lat: PropTypes.number.isRequired, 
  lng: PropTypes.number.isRequired, 
}

export default Marker