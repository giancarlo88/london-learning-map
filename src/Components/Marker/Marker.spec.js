import React from 'react'
import { shallow } from 'enzyme'

import Marker from './Marker'

describe(Marker.name, () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const props = {
    lat: 0,
    lng: 0,
    index: 0,
    handleClick: () => {}
  }

  describe('@renders', () => {
    it('should render', () => {
      const wrapper = shallow(<Marker {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('@props', () => {
    it('should fire handleClick prop when clicked', () => {
      const onClickMock = jest.fn()
      const wrapper = shallow(<Marker {...props} handleClick={onClickMock} />)
      wrapper.simulate('click')
      expect(onClickMock).toHaveBeenCalledTimes(1)
    })
  })
})
