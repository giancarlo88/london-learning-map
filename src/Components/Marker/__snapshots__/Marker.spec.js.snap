// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker @renders should render 1`] = `
<styled.div
  index={0}
  lat={0}
  lng={0}
  onClick={[Function]}
/>
`;
